pkgs/tools/misc/beam-book/default.nix

```diff
@@ -0,0 +1,98 @@
+{ lib, stdenv
+, fetchFromGitHub
+, wget, asciidoctor, ditaa, graphviz
+, gitMinimal
+, erlang
+, jdk
+, rsync
+, rouge
+}:
+
+let
+  pname = "the-beam-book";
+  version = "0.0.14.fix";
+
+  bookDirName = "${pname}-${version}";
+
+  book = fetchFromGitHub {
+    name = bookDirName;
+    owner = "happi";
+    repo = "theBeamBook";
+    rev = version;
+    sha256 = "06r19063a9s6hg84garn83rn4cnnkvs58x3j68zvf94mwsmfs86h";
+    leaveDotGit = true;
+  };
+
+  erlangGenOp = fetchFromGitHub {
+    name = "erlang.tar.gz";
+    owner = "erlang";
+    repo = "otp";
+    rev = "master";
+    sha256 = "1qfdry8wdba3abfsa1qkkcglww186j8dyl40si1vrwsc7f221r1d";
+  };
+
+in stdenv.mkDerivation rec {
+  inherit pname version;
+
+  srcs = [ book erlangGenOp ];
+
+  outputs = [ "out" "pdf" ];
+
+  sourceRoot = bookDirName;
+
+  postUnpack = ''
+    cp ${erlangGenOp}/lib/compiler/src/genop.tab ${bookDirName}/
+  '';
+
+  patches = [
+    # Cannot call wget in build environment. File is fetched during unpackPhase
+    ./remove-wget.patch
+
+    # bin/gitlog.sh script calls `git shortlog` without commit parameter, which
+    # is problematic in non-interactive shell sessions.
+    # https://stackoverflow.com/a/43042363
+    ./add-HEAD-to-git-shortlog.patch
+  ];
+
+  # Patch shebangs after we apply all patches
+  postPatch = ''
+    patchShebangs .
```

Comment:
Can we be more specific?

Reply:
More specific, as in more specific about which files to patch shebangs for? Like saying `patchShebangs ./path/to/file`, rather than doing a broad `patchShebangs .`?

Reply:
Yes, exactly.